sympy/solvers/diophantine.py

```diff
@@ -1040,13 +1040,13 @@ def _diop_quadratic(var, coeff, t):
 
         if D < 0:
             for solution in solns_pell:
-                s1 = P*Matrix([solution[0], solution[1]]) + Q
-                s2 = P*Matrix([-solution[0], solution[1]]) + Q
-                try:
-                    sol.add(tuple([as_int(_) for _ in s1]))
-                    sol.add(tuple([as_int(_) for _ in s2]))
-                except ValueError:
-                    pass
+                for X_i in [-solution[0], solution[0]]:
```

Comment:
It looks like `permute=True` will only handle the case where no linear terms in x or y are present. Hence something like this loop will be needed. But the code could be made more readable. For example,
`for solution in solns_pell:`
could probably be changed to
`for x0, y0 in solns_pell:`,
and then x0 and y0 could be used instead of solution[0] and solution[1] (and x, y instead of X_i and Y_i).